src/Google.Pubsub.V1/SubscriberClient.cs

```diff
@@ -0,0 +1,661 @@
+        /// </summary>
+        /// <returns>A new instance of the default SubscriberSettings.</returns>
+        public static SubscriberSettings GetDefault() => new SubscriberSettings();
+
+        /// <summary>
+        /// Creates a deep clone of this object, with all the same property values.
+        /// </summary>
+        /// <returns>A deep clone of this set of Subscriber settings.</returns>
+        public SubscriberSettings Clone() => CloneInto(new SubscriberSettings
+        {
+        });
+    }
+
+    /// <summary>
+    /// Subscriber client wrapper, for convinient use.
+    /// </summary>
+    public abstract partial class SubscriberClient
+    {
+        /// <summary>
+        /// Default service parameters for the Subscriber service.
+        /// </summary>
+        public static class ServiceDefaults
+        {
+            /// <summary>
+            /// The default Subscriber network host.
+            /// </summary>
+            /// <remarks>
+            /// The default Subscriber network host is "pubsub-experimental.googleapis.com".
+            /// </remarks>
+            public static string Host { get; } = "pubsub-experimental.googleapis.com";
+
+            /// <summary>
+            /// The default Subscriber network port.
+            /// </summary>
+            /// <remarks>
+            /// The default Subscriber network port is 443.
+            /// </remarks>
+            public static int Port { get; } = 443;
+
+            /// <summary>
+            /// The default Subscriber scopes
+            /// </summary>
+            /// <remarks>
+            /// The default Subscriber scopes are:
+            /// <list type="bullet">
+            /// <item><description>"https://www.googleapis.com/auth/pubsub"</description></item>
+            /// <item><description>"https://www.googleapis.com/auth/cloud-platform"</description></item>
+            /// </list>
+            /// </remarks>
+            public static ReadOnlyCollection<string> Scopes { get; } = new ReadOnlyCollection<string>(new[] {
+                "https://www.googleapis.com/auth/pubsub",
+                "https://www.googleapis.com/auth/cloud-platform",
+            });
+        }
+
+        /// <summary>
+        /// Get a new instance of the default <see cref="ServiceEndpointSettings"/>.
+        /// </summary>
+        /// <remarks>
+        /// The default service endpoint settings are:
+        /// <list type="bullet">
+        /// <item><description>Host "pubsub-experimental.googleapis.com"</description></item>
+        /// <item><description>Port 443</description></item>
+        /// </list>
+        /// </remarks>
+        public static ServiceEndpointSettings GetDefaultServiceApiSettings() => new ServiceEndpointSettings
+        {
+            Host = ServiceDefaults.Host,
+            Port = ServiceDefaults.Port,
+        };
+
+        private static async Task<SubscriberClient> CreateFromDefaultCredentialsUnsafeAsync(
```

Comment:
When this is committed, I'll have a go at refactoring to put most of this into the GAX.